### PR TITLE
SALTO-825: added more types to Netsuite

### PR DESCRIPTION
**Pull request by alonstern**
This PR includes:
- Adding support for more types in Netsuite
- Adding the types of more fields in Netsuite
- Adding support in fetching items

---
_Release Notes_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/data_elements/data_elements.ts [hunk 1]
```diff
@@ -120,13 +137,14 @@ export const getDataElements = async (
...
-          record => createInstance(record, typesMap[typeName])
-        ))))))
+  const availableTypesToFetch = typesToFetch.filter(typeName => typeName in typesMap)
+
+  if (availableTypesToFetch.length === 0) {
+    return types
+  }
+
+  const instances = await awu((await client.getAllRecords(availableTypesToFetch)))
+    .map(record => createInstance(record, typesMap)).toArray()
```
Comment (omrilit) [inline packages/netsuite-adapter/src/data_elements/data_elements.ts:147]: styiling nit:
```suggestion
  const instances = await awu(await client.getAllRecords(availableTypesToFetch))
  .map(record => createInstance(record, typesMap))
  .toArray()
```

File: packages/netsuite-adapter/src/data_elements/types.ts [hunk 2]
```diff
@@ -153,4 +170,47 @@ export const INTERNAL_ID_TO_TYPES: Record<string, string[]> = _(TYPES_TO_INTERNA
   .mapValues(values => values.map(([type]) => type))
   .value()
 
+export const ITEM_TYPE_ID = '-10'
+export const TRANSACTION_TYPE_ID = '-30'
```
Comment (omrilit) [inline packages/netsuite-adapter/src/data_elements/types.ts:174]: Do you think that it would be better to use these consts also in `TYPES_TO_INTERNAL_ID`?

**Review by omrilit — COMMENTED** [review]
Looks great!

2 general questions:
1. When a user will set `OtherChargeSaleItem` in `fetch.exclude`, will we really skip its fetch?
    I ask because for example it has the same searchString as `OtherChargeResaleItem`.
2. If fetching a certain type fails, will we fail the whole fetch operation? Do we have any config suggestion mechanism in place for the SuiteApp results?

File: packages/netsuite-adapter/src/data_elements/data_elements.ts [hunk 3]
```diff
@@ -67,10 +67,27 @@ export const getDataTypes = async (
...
 }
 
+const getType = (
+  record: Record<string, unknown>,
+  typesMap: Record<string, ObjectType>
+): ObjectType => {
+  const typeNames = Object.entries(
+    (record as { attributes?: Record<string, unknown> }).attributes ?? {}
+  ).filter(([key, value]) => key.split(':')[1] === 'type' && typeof value === 'string')
+    .map(([_key, value]) => (value as string).split(':')[1])
```
[the remark this answers is not in the document]
Comment (alonstern) [reply packages/netsuite-adapter/src/data_elements/data_elements.ts:77]: I think we can but just in case added a check

**Comment by alonstern** [comment]
> Looks great!
> 
> 2 general questions:
> 
> 1. When a user will set `OtherChargeSaleItem` in `fetch.exclude`, will we really skip its fetch?
>    I ask because for example it has the same searchString as `OtherChargeResaleItem`.
> 2. If fetching a certain type fails, will we fail the whole fetch operation? Do we have any config suggestion mechanism in place for the SuiteApp results?

1. Yes, it won't pass `OtherChargeSaleItem` to the `subtypes` so the items' search won't return `OtherChargeSaleItem`
2. Currently we don't. IMO we might want to wait with it to see what reasons could make the fetching of a type fail and if it makes sense to add the types to the skip list in those scenarios. (Also, since we request all the item types in one request I am not sure we have a way to know programmatically which type causes the request to fail)

File: packages/netsuite-adapter/src/data_elements/data_elements.ts [hunk 4]
```diff
@@ -144,7 +148,8 @@ export const getDataElements = async (
   }
 
   const instances = await awu((await client.getAllRecords(availableTypesToFetch)))
```
Comment (omrilit) [inline packages/netsuite-adapter/src/data_elements/data_elements.ts:150]: nit: redundant parenthesis

**Review by omrilit — APPROVED** [review]

**Pull request merged**